File: Leet-Code/FindFirstIndexOfString.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        haystack_length, needle_length = len(haystack), len(needle)
        last_possible_index = haystack_length - needle_length

        for index in range (0, last_possible_index + 1):
            if haystack[index: index + needle_length] == needle:
                return index
        return -1

File: Leet-Code/FindFirstIndexOfString.cpp
class Solution {
public:
    int strStr(string haystack, string needle) {
        const int haystackLength = haystack.size();
        const int needleLength = needle.size();

        for(int i = 0; i <= haystackLength - needleLength; ++i)
        {
            if(haystack.substr(i, needleLength) == needle)
                return i;
        }
        return -1;
    }
};
